tasks/github_tasks.py

```diff
@@ -188,3 +189,60 @@ def get_token_from_app(_, app_id_env='GITHUB_APP_ID', pkey_env='GITHUB_KEY_B64')
     from .libs.common.github_api import GithubAPI
 
     GithubAPI.get_token_from_app(app_id_env, pkey_env)
+
+
+@task
+def assign_team_label(_, pr_id=-1):
+    """
+    Assigns the github team label name if a single team can
+    be deduced from the changed files
+    """
+    from .libs.common.github_api import GithubAPI
+
+    def get_team(changed_files) -> str | None:
```

Comment:
I would maybe tend to do the other way around, ie have `get_team` as a service of GithubAPI with codeowners as input. It's then maybe easier to do a unit test of this service, wdyt?

Reply:
I isolated this function, tell me if it is the proper way